From a94922b8306623bb6d0f8c50105df889461cac0f Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 9 May 2019 22:40:04 +0200
Subject: [PATCH 178/345] MIPS: Add new function local_flush_icache_all()

... to flush the entire local ICache.  Default implementation is to do
flush_icache_all().  Processor specific implementations will be in
follow on patches.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/include/asm/cacheflush.h | 1 +
 arch/mips/mm/cache.c               | 5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/mips/include/asm/cacheflush.h b/arch/mips/include/asm/cacheflush.h
index 4812d1fed0c2..0d30b4b0d98b 100644
--- a/arch/mips/include/asm/cacheflush.h
+++ b/arch/mips/include/asm/cacheflush.h
@@ -112,6 +112,7 @@ extern void copy_from_user_page(struct vm_area_struct *vma,
 
 extern void (*flush_cache_sigtramp)(unsigned long addr);
 extern void (*flush_icache_all)(void);
+extern void (*local_flush_icache_all)(void);
 extern void (*local_flush_data_cache_page)(void * addr);
 extern void (*flush_data_cache_page)(unsigned long addr);
 
diff --git a/arch/mips/mm/cache.c b/arch/mips/mm/cache.c
index 66ff5ada3240..768ab75b081e 100644
--- a/arch/mips/mm/cache.c
+++ b/arch/mips/mm/cache.c
@@ -51,6 +51,8 @@ void (*flush_cache_sigtramp)(unsigned long addr);
 void (*local_flush_data_cache_page)(void * addr);
 void (*flush_data_cache_page)(unsigned long addr);
 void (*flush_icache_all)(void);
+void (*local_flush_icache_all)(void);
+EXPORT_SYMBOL(local_flush_icache_all);
 
 EXPORT_SYMBOL_GPL(local_flush_data_cache_page);
 EXPORT_SYMBOL(flush_data_cache_page);
@@ -240,7 +242,8 @@ void cpu_cache_init(void)
 
 		octeon_cache_init();
 	}
-
+	if (!local_flush_icache_all)
+		local_flush_icache_all = flush_icache_all;
 	setup_protection_map();
 }
 
-- 
2.25.1

